controllers/chat/sendAnonymousMessage.js

```diff
@@ -80,6 +83,11 @@ const sendAnonymousMessage = async (req, res) => {
       if (isBlocked) {
         return res.status(403).json(responseError('This user does not want to receive messages'));
       }
+      channelMember = better_channel_member.filter(
+        (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
+      );
+
+      senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The filtering logic for `channelMember` and assignment of `senderInfo` should be done outside the `if` block. This ensures that these operations are performed regardless of whether the user is blocked or not, which seems to be the intended behavior based on the context provided.
```diff
-      channelMember = better_channel_member.filter(
-        (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
-      );
-
-      senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
    }
 
+   channelMembers = better_channel_member.filter(
+     (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
+   );
+
+   senderInfo = better_channel_member.find((element) => element.user_id === req.userId);

    const currentMessageType = determineMessageType(messageType, attachments);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->